commit 2b9aa337af7291d3f141322da96c9f667c99d53c
Author: NeilBrown <neilb@suse.de>
Date:   Thu Oct 1 12:51:04 2009 +1000

    Fix null-dereference in set_member_info
    
    set_member_info would try to dereference ->metadata_version, without
    checking that it isn't NULL.
    
    Signed-off-by: NeilBrown <neilb@suse.de>

diff --git a/mapfile.c b/mapfile.c
index a3038be..ed59db5 100644
--- a/mapfile.c
+++ b/mapfile.c
@@ -303,19 +303,22 @@ struct map_ent *map_by_name(struct map_ent **map, char *name)
  */
 static void set_member_info(struct supertype *st, struct mdstat_ent *ent)
 {
-	char version[strlen(ent->metadata_version)+1];
 
 	st->subarray[0] = '\0';
 
-	if (strncmp(ent->metadata_version, "external:", 9) != 0)
+	if (ent->metadata_version == NULL ||
+	    strncmp(ent->metadata_version, "external:", 9) != 0)
 		return;
 
-	strcpy(version, ent->metadata_version);
-
-	if (is_subarray(&version[9])) {
-		char *subarray = strrchr(version, '/');
+	if (is_subarray(&ent->metadata_version[9])) {
+		char version[strlen(ent->metadata_version)+1];
+		char *subarray;
 		char *name = &version[10];
 
+		strcpy(version, ent->metadata_version);
+		subarray = strrchr(version, '/');
+		name = &version[10];
+
 		if (!subarray)
 			return;
 		*subarray++ = '\0';

commit 0e90271e53c0f6efb33e904cf407498350e2a14d
Author: NeilBrown <neilb@suse.de>
Date:   Thu Oct 1 12:38:31 2009 +1000

    Add missing space in "--detail --brief" output.
    
    We need a space between the device name and the word "level"..
    
    Signed-off-by: NeilBrown <neilb@suse.de>

diff --git a/Detail.c b/Detail.c
index 001012a..a70db34 100644
--- a/Detail.c
+++ b/Detail.c
@@ -207,11 +207,11 @@ int Detail(char *dev, int brief, int export, int test, char *homehost)
 		printf("ARRAY %s", dev);
 		if (brief > 1) {
 			if (array.raid_disks)
-				printf("level=%s num-devices=%d",
+				printf(" level=%s num-devices=%d",
 				       c?c:"-unknown-",
 				       array.raid_disks );
 			else
-				printf("level=container num-devices=%d",
+				printf(" level=container num-devices=%d",
 				       array.nr_disks);
 		}
 		if (container) {
